wordpress-mobile/WordPress-Android
# Feature/master post filters 9184

Fixes #9184 

This change adds the ability to view the list of blog posts in a compact view with a toggle in the top right hand side of the screen, to move between the new compact view and the standard view which is already present in the application. The new view has only a title, the date of the article and a list of statuses shown, as well as a much smaller image. All of the actions are hidden behind a menu using a vertical ellipsis on the right hand side of the row. The view type is persisted so that on returning to the view it remains the same.

To test:

1. Open the app and open blog posts
2. Make sure there are visible blog posts within the current section
3. Hit the button in the top right and check that the view displays as the compacted version (check against the design in the issue #9184) 
4. Hit the button again and make sure the view displays as the original version again
5. Hit the button one more time to return the view to compact
6. Kill the app (swipe out of multi-tasking)
7. Return to the app and make sure the view still displays as compact
8. Open the menu on the right hand side of the compact view, and make sure all button options are present, as they would be on the non-compacted versions.

Update release notes:

- [Y] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
